### PR TITLE
Stripe integration

### PR DESCRIPTION
**Pull request by th0rgall**

### CODE REVIEW

**Review by wardbeyens — CHANGES_REQUESTED** [review]

File: src/routes/become-superfan/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+<script lang="ts">
+  import { _ } from 'svelte-i18n';
+  import { user } from '@/lib/stores/auth';
+  import { goto } from '$app/navigation';
+  import enterHandler from '@/lib/util/keyhandlers';
+  import SuperfanLevel from '@/lib/components/UI/SuperfanLevel.svelte';
+  import routes from '@/lib/routes';
+
+  function selectLevel(level: SuperfanLevelData) {
```
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/routes/become-superfan/+page.svelte:38]: Noted, will change!

File: src/routes/become-superfan/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      stripePriceId: 'price_1MAcGbDY48WHyD7GY6TCMmoC'
+    },
+    {
+      title: 'Solidarity price',
+      id: 'solidarity',
+      value: 7,
+      stripePriceId: 'price_1MAcGzDY48WHyD7GVUZAbVcf'
+    }
+  ];
+</script>
```
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/routes/become-superfan/+page.svelte:28]: Good question. This whole `become-superfan/+page.svelte` and `become-superfan/[id]/+page.svelte` is some messy copy-paste stuff.

I can imagine these levels can be useful in different parts of the app too, I'll try to move them somewhere to `lib`.

File: src/routes/become-superfan/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  import { _ } from 'svelte-i18n';
+  import { user } from '@/lib/stores/auth';
+  import { goto } from '$app/navigation';
+  import enterHandler from '@/lib/util/keyhandlers';
+  import SuperfanLevel from '@/lib/components/UI/SuperfanLevel.svelte';
+  import routes from '@/lib/routes';
+
+  function selectLevel(level: SuperfanLevelData) {
+    if (!user) {
+      return goto(routes.SIGN_IN);
```
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/routes/become-superfan/+page.svelte:40]: Good idea!

File: src/routes/become-superfan/+page.svelte [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    const handler = enterHandler(() => selectLevel(item));
+    handler(event.detail);
+  }
+</script>
+
+<svelte:head>
+  <title>{$_('account.title')} | Welcome To My Garden</title>
+</svelte:head>
+
+<!-- TODO: find a way to generalize the wrapper across pages -->
```
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/routes/become-superfan/+page.svelte:56]: Yep, I got the hang of slots yesterday, I already fixed this :)

File: src/routes/become-superfan/payment/+page.svelte [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  let unsubscribeFromUser = user.subscribe((newUser) => {
+    if (newUser) selectLevelFromSubscription(newUser);
+  });
+
+  onMount(async () => {
+    // todo: validate priceID
+    if (!$user) {
+      return goto(routes.SIGN_IN);
```
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/routes/become-superfan/payment/+page.svelte:40]: What do you mean with a guard?

File: src/routes/become-superfan/payment/+page.svelte [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  })
+</script>
+
+<svelte:head>
+  <title>{$_('account.title')} | Welcome To My Garden</title>
+</svelte:head>
+
+<!-- TODO: find a way to generalize the wrapper across pages -->
+<div class="wrapper">
+  {#if $user && hasActiveSubscription($user)}
```
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/routes/become-superfan/payment/+page.svelte:66]: If I understood the docs correctly, I think you're wrong on this one :) 

https://svelte.dev/docs#component-format-script-4-prefix-stores-with-$-to-access-their-values the docs don't mention that this doesn't work in templates
> Any time you have a reference to a store, you can access its value inside a component by prefixing it with the $ character. This causes Svelte to declare the prefixed variable, subscribe to the store at component initialization and unsubscribe when appropriate.
https://svelte.dev/docs#component-format-script-4-prefix-stores-with-$-to-access-their-values

And in this example, reactivity works in `App.svelte` without using $count in the <script>: https://svelte.dev/tutorial/auto-subscriptions

File: src/lib/components/UI/SuperfanLevel.svelte [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<script lang="ts">
```
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/lib/components/UI/SuperfanLevel.svelte:1]: Probably going to move this to `src/routes/(marketing)/_components`, if it's not going to be used elsewhere in the main app.

File: src/routes/become-superfan/payment/+page.svelte (hunk 5 above)
[the remark this answers is not in the document]
Comment (wardbeyens) [reply src/routes/become-superfan/payment/+page.svelte:40]: https://svelte.dev/repl/5673ff403af14411b0cd1785be3d996f?version=3.55.0

File: src/lib/components/UI/SuperfanLevel.svelte (hunk 7 above)
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/lib/components/UI/SuperfanLevel.svelte:1]: > add component in /UI/index.js

I actually noticed this today several times: when opening import suggestions, I can often choose between the index.js export and the actual component export. What do you think is the benefit of having an index file?

File: src/routes/become-superfan/payment/+page.svelte (hunk 5 above)
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/routes/become-superfan/payment/+page.svelte]: Hmm, that auth guard is just an alias for the same if statement that's there now. I think it's actually more simple with the if statement, with the auth guard you need to remember which slot names to use.

The UX here is not good though (client-side redirect, without warning). Should be improved.

File: src/routes/become-superfan/+page.svelte (hunk 3 above)
[the remark this answers is not in the document]
Comment (th0rgall) [reply src/routes/become-superfan/+page.svelte]: Going to close this to move on and merge branches (clean up), the task is in Trello here: https://trello.com/c/EwHqiLi2/2805-superfan-payment-page

**Pull request merged**